### ISSUE TITLE
Official release of Perl 6 implementation

### ISSUE THREAD

**Issue by rba** [perl6/problem-solving#70]
**I would like to have an official language release with all of it needed components for coding in Perl 6.**

I know that the spectest is it, yet this is hard to use. And I don't forget that there are multiple compiler implementations.

For a developer in a company it's important that he know what he can get. So having the latest Perl 6 installed with features x, y and z.

For this I would recommend to use Rakudo Star and then make the @nxadm built packages official. Something like a quick install for Linux, à la pull https://metacpan.org/pod/App::cpanminus or https://www.rust-lang.org/learn/get-started, where a script is pulled and runned, would be helpful for one who likes to have quick look.

To sum up, I like to have:

- A Perl 6 official release with all it's components, ready to code
- Packages for multiple platforms based on the official release.
- An optional script installer, which pull binaries

For myself I like to have an Red Hat Enterprise Linux (RHEL) package.

Open for discussion.

**Comment by nxadm** [comment]
Now that Rakudo is relocatable, I am looking at creating compile tar.gz like GoLang does. Just an option.

**Comment by AlexDaniel** [comment]
Related: https://github.com/perl6/user-experience/issues/29

**Comment by patzim** [comment]
Related: [rakudo/star#140](https://github.com/rakudo/star/issues/140)

**Comment by patzim** [comment]
There is also [rakudo/rakudo#PR3049](https://github.com/rakudo/rakudo/pull/3049). That's the start of a release manual on how to create binary zip/tgz releases. Covered in those docs are Windows and Linux at the moment.

**Comment by patzim** [comment]
The binary release doc PR is merged. Nothing in there is set in stone and the guide needs to be extended to also cover MacOS and other platforms we want to commit to releasing for. (Raspi comes to mind)

**Comment by jnthn** [comment]
> A Perl 6 official release

I don't particularly think we need to start sticking the label "official" anywhere, so much as to make sure that we provide a very clear way for folks to get a Perl 6 implementation on their computer so they can Do Stuff, without having to first understand what lots of different things (Rakudo, Rakudo Star, MoarVM, zef, etc.) are. Instead, we'd offer options like "Install Perl 6 compiler and module manager using apt", "Install a Perl 6 compiler by running a script", etc.

> with all it's components, ready to code

I suspect expectations on what "all its components" mean will vary quite widely. At $dayjob we're thinking a bit about whether to put together a single download/installer where you get all(Rakudo, Cro, Comma IDE), which would be very much "all components", but probably not what everyone wants. :-)

Maybe we need some very simple "wizard" process, that starts with "OK, I see you're on [operating system]" and proposes a few simple options. For example: if it detects Windows, it could offer an MSI or a ZIP (and state - in generic terms - what you get in them).

**Comment by nxadm** [comment]
> Maybe we need some very simple "wizard" process, that starts with "OK, I see you're on [operating system]" and proposes a few simple options. For example: if it detects Windows, it could offer an MSI or a ZIP (and state - in generic terms - what you get in them).

Whatever is decided for the other points in the discussion, please -please- no wizards. Wizards are not only an hassle, they are more often than not wrong. E.g., what if an admin that is forced to used MS Windows at work is looking for Linux binaries? What a Windows user would like a portable app install (because he/she does not have write permissions elsewhere? And so on.

An example of a download page I like because it's functional:
https://golang.org/dl/

**Comment by jnthn** [comment]
> Whatever is decided for the other points in the discussion, please -please- no wizards

Apparently putting "wizard" in quotes wasn't enough. :-) The point was simply to have an interface where you can choose an OS (defaulting to the one you're currently on), and then choose from a set of common options that are suitable for that.

> E.g., what if an admin that is forced to used MS Windows at work is looking for Linux binaries?

Then you'd just click on Linux and see the options for that.

> What a Windows user would like a portable app install (because he/she does not have write permissions elsewhere?

Then that'd be one of the options presented for Windows, presumably alongside an MSI.

**Comment by cup** [comment]
https://github.com/MoarVM/MoarVM/issues/1124

**Comment by cup** [comment]
@nxadm are you sure about that? It seems #1124 is still open

**Comment by nxadm** [comment]
> @nxadm are you sure about that? It seems [MoarVM/MoarVM#1124](https://github.com/MoarVM/MoarVM/issues/1124) is still open

Yes, I provide relocatable Linux builds:
https://github.com/nxadm/rakudo-pkg/releases

**Comment by cup** [comment]
Linux is not the only operating system supported by Raku...

**Comment by nxadm** [comment]
No, but as far as I know, the relocatable part does work on other OSes (I am not sure about the mentioned problem with spaces on Windows). I am considering adding Windows and MacOS packages to rakudo-pkg but I haven't had the time.

**Comment by cup** [comment]
@nxadm no, it definitely doesnt. The current Windows build is not relocatable.

**Comment by nxadm** [comment]
@cup, great to know, so I know it makes no sense for me to make Windows packages at the moment. I had done some work but still in the struggling stage of unix-guy-trying-to-create-a-working-windows-development-environment-on-Travis. ;)
